File: JavaPractice/Hello.java
class Hello {
  public static void main(String[] args) {
    System.out.println("こんにちは");
  }
}


// 【main メソッドについて】
// 簡単に言えば main メソッドは以下の通りそのまま書けばいいだけです。

// public static void main(String[] args) {
// }

// もちろんそれぞれに意味はありますので、簡単に説明します。

// (1)main メソッドは public キーワードをつけて宣言します。public は他の全てのクラスから呼び出せるということです。

// (2)main メソッドは static キーワードをつけます。static を付けるとクラスをインスタンス化しなくてもこのメソッドが使えます。

// (3)main メソッドは、戻り値に void キーワードを使います。void はそのメソッドが何も値を返さないことを意味します。

// (4)main メソッドは、引数としてString型の配列を受け取れるようにします。 args は配列の名前です。

File: NodePractice/hello.js
//console.log("Hello Node!");

// non blocking
//次の処理 (Worldを出力)を邪魔しないようにコールバック関数で最初の処理を渡す必要がある。
// setTimeout(function() {
//     console.log("hello");
// }, 2000);

// console.log("World");

//逆にブロックしているような処理
//Nodeはメインのスレッドが一つなので、この書き方をするとすべてのリクエストをブロックするというおそろしい事態になる
var start = new Date().getTime();
while(new Date().getTime() < start + 1000);
console.log("world");

//時間がかかりそうな処理はコールバック関数で実装すること